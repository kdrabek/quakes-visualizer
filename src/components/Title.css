.header {
    grid-column: 1 / 3;
    grid-row: 1 / 2;
    padding: 5px;
  }
  