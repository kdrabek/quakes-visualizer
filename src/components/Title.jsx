import IconButton from 'material-ui/IconButton';
import classNames from 'classnames/bind';
import { PropTypes } from 'prop-types';
import { Typography } from 'material-ui';

import styles from './Title.css';

const cx = classNames.bind(styles);

const header = cx({ header: true, 'flex-container': true });
const avatar = cx({ avatar: true });
const gitlab = cx({ fa: true, 'fa-gitlab': true });

const propTypes = PropTypes && {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
const defaultProps = {};


const Title = ({ text, url }) => (
  <div className={header}>
    <Typography variant="title">
      {text}
    </Typography>
    <span className={avatar}>
      <IconButton variant="fab" href={url}>
        <i className={gitlab} />
      </IconButton>
    </span>
  </div>
);

Title.displayName = 'Title';
Title.propTypes = propTypes;
Title.defaultProps = defaultProps;
export default Title;
