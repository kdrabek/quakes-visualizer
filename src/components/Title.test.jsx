import { shallow } from 'enzyme';
import Title from './Title';


describe('Title component', () => {
  const props = {
    text: 'some text',
    url: 'http://some.link',
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<Title {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
