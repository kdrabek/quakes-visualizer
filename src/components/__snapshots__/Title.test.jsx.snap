// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should match snapshot 1`] = `
<div
  className="header flex-container"
>
  <WithStyles(Typography)
    variant="title"
  >
    some text
  </WithStyles(Typography)>
  <span
    className="avatar"
  >
    <WithStyles(IconButton)
      variant="fab"
    >
      <i
        className="fa fa-gitlab"
      />
    </WithStyles(IconButton)>
  </span>
</div>
`;
